base/random/normal.jl

```diff
@@ -1,848 +1,208 @@
-@inline reserve(r::MersenneTwister, n::Int) = (mt_avail(r) < n && gen_rand(r); nothing)
-
-# precondition: !mt_empty(r)
-@inline rand_inbounds(r::MersenneTwister, ::Type{Close1Open2}) = mt_pop!(r)
-@inline rand_inbounds(r::MersenneTwister, ::Type{CloseOpen}) = rand_inbounds(r, Close1Open2) - 1.0
-@inline rand_inbounds(r::MersenneTwister) = rand_inbounds(r, CloseOpen)
-
-# produce Float64 values
-@inline rand(r::MersenneTwister, ::Type{I}) where {I<:FloatInterval} = (reserve_1(r); rand_inbounds(r, I))
-
-@inline rand_ui52_raw_inbounds(r::MersenneTwister) = reinterpret(UInt64, rand_inbounds(r, Close1Open2))
-@inline rand_ui52_raw(r::MersenneTwister) = (reserve_1(r); rand_ui52_raw_inbounds(r))
-@inline rand_ui2x52_raw(r::MersenneTwister) = rand_ui52_raw(r) % UInt128 << 64 | rand_ui52_raw(r)
-
-function srand(r::MersenneTwister, seed::Vector{UInt32})
-    copy!(resize!(r.seed, length(seed)), seed)
-    dsfmt_init_by_array(r.state, r.seed)
-    mt_setempty!(r)
-    return r
-end
-
-# MersenneTwister jump
-
-"""
-    randjump(r::MersenneTwister, jumps::Integer, [jumppoly::AbstractString=dSFMT.JPOLY1e21]) -> Vector{MersenneTwister}
-
-Create an array of the size `jumps` of initialized `MersenneTwister` RNG objects. The
-first RNG object given as a parameter and following `MersenneTwister` RNGs in the array are
-initialized such that a state of the RNG object in the array would be moved forward (without
-generating numbers) from a previous RNG object array element on a particular number of steps
-encoded by the jump polynomial `jumppoly`.
-
-Default jump polynomial moves forward `MersenneTwister` RNG state by `10^20` steps.
-"""
-function randjump(mt::MersenneTwister, jumps::Integer, jumppoly::AbstractString)
-    mts = MersenneTwister[]
-    push!(mts, mt)
-    for i in 1:jumps-1
-        cmt = mts[end]
-        push!(mts, MersenneTwister(copy(cmt.seed), dSFMT.dsfmt_jump(cmt.state, jumppoly)))
+@inline function randn(rng::AbstractRNG=GLOBAL_RNG)
+    @inbounds begin
+        r = rand_ui52(rng)
+        rabs = Int64(r>>1) # One bit for the sign
+        idx = rabs & 0xFF
+        x = ifelse(r % Bool, -rabs, rabs)*wi[idx+1]
+        rabs < ki[idx+1] && return x # 99.3% of the time we return here 1st try
+        return randn_unlikely(rng, idx, rabs, x)
     end
-    return mts
 end
-randjump(r::MersenneTwister, jumps::Integer) = randjump(r, jumps, dSFMT.JPOLY1e21)
 
-## initialization
-
-function __init__()
-    try
-        srand()
-    catch ex
-        Base.showerror_nostdio(ex,
-            "WARNING: Error during initialization of module Random")
+# this unlikely branch is put in a separate function for better efficiency
+function randn_unlikely(rng, idx, rabs, x)
+    @inbounds if idx == 0
+        while true
+            xx = -ziggurat_nor_inv_r*log(rand(rng))
+            yy = -log(rand(rng))
+            yy+yy > xx*xx && return (rabs >> 8) % Bool ?
+                -ziggurat_nor_r - xx :
+                ziggurat_nor_r  + xx
```

Comment:
this would probably be clearer written out longhand as
```
            if yy+yy > xx*xx
                if (rabs >> 8) % Bool
                    return -ziggurat_nor_r - xx
                else
                    return ziggurat_nor_r  + xx
                end
            end
```